src/background.js

```diff
@@ -1 +1,38 @@
-/* background.js */
+import ExtensionMonitor from './lib/ext-monitor.js';
+
+const extensionMonitor = new ExtensionMonitor();
+
+function messageHandler(message, sender, sendResponse) {
+  const {
+    extStartMonitorAllExts,
+    extStopMonitorAll,
+    sendAllExistingLogs,
+    getMonitoringStatus,
+  } = message;
+
+  if (getMonitoringStatus) {
+    sendResponse({
+      status: extensionMonitor.areExtsBeingMonitored(),
+    });
+  }
+
+  if (Array.isArray(extStartMonitorAllExts)) {
+    return extensionMonitor
+      .initMonitor(extStartMonitorAllExts)
+      .then((monitorMsg) => monitorMsg);
+  }
+
+  if (extStopMonitorAll) {
+    return extensionMonitor.stopMonitorAll().then((monitorMsg) => monitorMsg);
+  }
+
+  // This is send all existing logs to activitylogs page
+  // once activitylogs page is initialized
+  if (sendAllExistingLogs) {
+    sendResponse({
+      existingLogs: extensionMonitor.logs,
+    });
+  }
```

Comment:
As I did mention in another review comment, these runtime.onMessage events could be better represented by messages in the form:
```
{
  requestType: "...", // where requestType could be "getMonitorStatus", "startMonitor", "stopMonitor" ...
  args: [...], // where args is an array (or an object if you prefer) of arguments for the specific requestType.
}
```

This would:
- make it clear that every one of this messages handles just a single request type
- the message handler could easily delegate the request type to a "requestType"-specific method based on just the `requestType` property.

Reply:
It doesn't have to be called `args`. The most important part is that using e.g. `requestType` would make it very clear that a message is expected to be handled by one handler only.

There are multiple ways to handle those messages, e.g. `switch`-`case` / `if`-`else if` / putting message handlers on an object (e.g. as seen here: https://github.com/mozilla/FirefoxColor/blob/9d27d0d10249f8b20d36f629b6c9cd7ea14532f0/src/extension/background.js#L46-L108)

Reply:
Thanks for this idea, I have implemented it.